### PR TITLE
Interface for sending SMS to the data collector

### PR DESCRIPTION
**Pull request by cathinkaw**
**Interface for sending SMS to the data collector**

```
    Summary of the changes
    - filtering data collectors
    - A box for the text to be sent to the filtered data collectors 
    - a button to send the manually entered text to the data collectors selected by SMS
    Fixes #339
```

### CODE REVIEW

File: Source/Reporting/Web/src/app/Management/DataCollectors/send-sms/send-sms.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+  public error: boolean;
+  public errorMsg = 'Could not get users, try again later';
+
+  selectedUser: DataCollector;
+  itemList;
+  selectedItems = [];
+  settings = {};
+  message = '';
+  m
```
Comment (cathinkaw) [inline Source/Reporting/Web/src/app/Management/DataCollectors/send-sms/send-sms.ts]: what is the m?

File: Source/Reporting/Web/src/app/Management/DataCollectors/send-sms/send-sms.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  recipients;
+
+  constructor(
+    private queryCoordinator: QueryCoordinator
+  ) { }
+
+  ngOnInit() {
+    this.queryCoordinator.execute(new AllDataCollectors())
```
Comment (cathinkaw) [inline Source/Reporting/Web/src/app/Management/DataCollectors/send-sms/send-sms.ts:22]: Great that you have just solved the lack of backend with some console logs.

File: Source/Reporting/Web/src/styles.scss [hunk 3]
```diff
@@ -20,3 +20,127 @@ form { // TODO: Style the user-form with properly scoped CSS
...
+
+
+
+$base-color: #0079FE;
+$btn-background: #fff;
+$btn-border: #ccc;
+$btn-text-color: #333;
+$btn-arrow: #333;
+
+
```
Comment (cathinkaw) [inline Source/Reporting/Web/src/styles.scss:33]: Remove one of the new lines

File: Source/Reporting/Web/src/app/Management/DataCollectors/send-sms/send-sms.html [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      <label for="full_name">Recipients</label>
+      <angular2-multiselect name="recipients" [data]="itemList" [(ngModel)]="selectedItems" [settings]="settings" (onSelect)="onItemSelect($event)" (onDeSelect)="OnItemDeSelect($event)" (onSelectAll)="onSelectAll($event)" (onDeSelectAll)="onDeSelectAll($event)">
+        <c-badge>
+          <ng-template let-item="item">
+              <label style="margin: 0px;">{{item.fullName}}</label>
+          </ng-template>
+        </c-badge>
+        <c-item>
+          <ng-template let-item="item">
+            <label style="color: #333;min-width: 150px;">{{item.fullName +' ' + item.phoneNumbers}}</label>
```
Comment (cathinkaw) [inline Source/Reporting/Web/src/app/Management/DataCollectors/send-sms/send-sms.html]: We sholdn't have styling inline. Maybe you can move it to the send-sms.scss file?

**Review by cathinkaw — COMMENTED** [review]
Great work, @VihoSerge ! I made a new pull request in our own fork as we are having some issued with how we should do the pull requests into the IRFCGo-fork (as there are many other changes coming with).

Se 3 small comments. If you don't move the styling to sms-send.scss that file should be deleted :)

File: Source/Reporting/Web/src/app/Management/DataCollectors/send-sms/send-sms.html (hunk 4 above)
Comment (cathinkaw) [reply Source/Reporting/Web/src/app/Management/DataCollectors/send-sms/send-sms.html]: There is also some inline styling on the label that is a bit higher in the file.

**Review by cathinkaw — COMMENTED** [review]
Great! I will wait to merge it because we don't have the backend support yet. But great work @VihoSerge  :D

**Comment by woksin** [comment]
I merged it with master and fixed an error that I couldn't run the frontend because of a version mismatch. However this SMS interface does not work fully as intended I guess, that may be because I merged with master.

@VihoSerge @cathinkaw

**Review by woksin — CHANGES_REQUESTED** [review]
Merge with master and make it functional

**Pull request closed without merging**